test/util_midgard.cc

```diff
@@ -459,6 +459,27 @@ TEST(UtilMidgard, TrimShapeEmpty) {
   ASSERT_EQ(shape.size(), 0);
 }
 
+// Test cases from: https://tools.ietf.org/html/rfc4648#section-10
```

Comment:
@kevinkreiser I moved the base64 encoding stuff into a header and added some test cases from the base64 encoding RFC and ran into some null termination issues. I expected roundtrip equality here, but It looks like we're adding some extra bytes here? 

```
/Users/buromookerji/projects/mapbox/valhalla/test/util_midgard.cc:478: Failure
Expected equality of these values:
  decode64(encoded)
    Which is: "f\0\0"
  decoded
    Which is: "f"
/Users/buromookerji/projects/mapbox/valhalla/test/util_midgard.cc:479: Failure
Expected equality of these values:
  decode64(encode64(decoded))
    Which is: "f\0\0"
  decoded
    Which is: "f"
/Users/buromookerji/projects/mapbox/valhalla/test/util_midgard.cc:478: Failure
Expected equality of these values:
  decode64(encoded)
    Which is: "fo\0"
  decoded
    Which is: "fo"
/Users/buromookerji/projects/mapbox/valhalla/test/util_midgard.cc:479: Failure
Expected equality of these values:
  decode64(encode64(decoded))
    Which is: "fo\0"
  decoded
    Which is: "fo"
/Users/buromookerji/projects/mapbox/valhalla/test/util_midgard.cc:478: Failure
Expected equality of these values:
  decode64(encoded)
    Which is: "foob\0\0"
  decoded
    Which is: "foob"
/Users/buromookerji/projects/mapbox/valhalla/test/util_midgard.cc:479: Failure
Expected equality of these values:
  decode64(encode64(decoded))
    Which is: "foob\0\0"
  decoded
    Which is: "foob"
/Users/buromookerji/projects/mapbox/valhalla/test/util_midgard.cc:478: Failure
Expected equality of these values:
  decode64(encoded)
    Which is: "fooba\0"
  decoded
    Which is: "fooba"
/Users/buromookerji/projects/mapbox/valhalla/test/util_midgard.cc:479: Failure
Expected equality of these values:
  decode64(encode64(decoded))
    Which is: "fooba\0"
  decoded
    Which is: "fooba"
```

Reply:
Fixed